node/impl/full/eth.go

```diff
@@ -570,12 +570,68 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 	return &receipt, nil
 }
 
-func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(context.Context, ethtypes.EthHash, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(ctx context.Context, blkHash ethtypes.EthHash, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	if err != nil {
+		if err == ErrNullRound {
+			return nil, xerrors.Errorf("block hash %s was a null round", blkHash)
+		}
```

Comment:
```suggestion
```

See below, let's address these all as part of https://github.com/filecoin-project/lotus/issues/12633